ddtrace/contrib/internal/openai/utils.py

```diff
@@ -429,25 +479,50 @@ def _compute_completion_tokens(completions_or_messages, model_name):
     return estimated, num_completion_tokens
 
 
-def _tag_tool_calls(integration, span, tool_calls, choice_idx):
+def _tag_tool_calls(integration, span, tool_calls, choice_idx=None):
     # type: (...) -> None
     """
-    Tagging logic if function_call or tool_calls are provided in the chat response.
+    Tagging logic if function_call, tool_calls in the chat response, or tools are provided in the responses.
     Notes:
         - since function calls are deprecated and will be replaced with tool calls, apply the same tagging logic/schema.
         - streamed responses are processed and collected as dictionaries rather than objects,
           so we need to handle both ways of accessing values.
     """
+    # Handle response tools
+    if hasattr(tool_calls, "tools") and tool_calls.tools != []:
```

Comment:
```suggestion
    tools = getattr(tool_calls, "tools", [])
    if tools and isinstance(tools, list):
```
This is a useful shorthand to check if something exists, is tool_calls.tools not empty, and make sure that it is a list